src/main/java/com/uber/kafkaSpraynozzle/KafkaPoster.java

```diff
@@ -16,61 +23,102 @@
     PoolingHttpClientConnectionManager cm;
     List<String> urls;
     static int currentUrl = 0;
+    static final int responseDecayingHalflife = 5;
     ConcurrentLinkedQueue<String> logQueue;
     KafkaFilter messageFilter;
+    private boolean roundRobin = false;
+    private int batchSize = 20;
+    private int batchMaxBytes = 128*1024;
+    private boolean debuggingOutput = false;
 
     public KafkaPoster(
+            ConcurrentLinkedQueue<ByteArrayEntity> queue,
+            PoolingHttpClientConnectionManager cm,
+            List<String> urls,
+            ConcurrentLinkedQueue<String> logQueue,
+            KafkaFilter messageFilter
+            ){
+        this(queue, cm, urls, logQueue, messageFilter, 20, false, false );
+    }
+    public KafkaPoster(
```

Comment:
Create a new constructor, so we can control whether we use round-robin to post, or print extra debugging output, or control batch size.